### PR TITLE
Add some documentation about debugging locally

### PR DESCRIPTION
**Pull request by diversemix**

### CODE REVIEW

File: local-development.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Local Deployment Tesing
+
+At times it may be necessary to change the way the deployment works, the most convienent way of doing this is to make changes locally. This avoids having to setup and wait for external services. This file contains some details on how to do this development locally.
```
Comment (giorgiosironi) [inline local-development.md]: `convenient`?

File: local-development.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+# Requirements
+
+You will need installed `Vagrant` and `VirtualBox`
+
+# Getting Started
+
+Clone the repoistory [builder](https://github.com/elifesciences/builder) onto your local machine:
```
Comment (giorgiosironi) [inline local-development.md]: `repository`

File: local-development.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+```
+  git clone git@github.com:elifesciences/builder.git
+```
+
+In the root of the project folder run `vagrant up` and select elife-xpub. Alternatively you can supply this in the `PROJECT` environment variable:
```
Comment (giorgiosironi) [inline local-development.md:18]: A link to https://github.com/elifesciences/builder/#vagrant will be authoritative on this, and get updated.

File: local-development.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+```
+    cloned-projects/elife-xpub-formula
+```
+
+Check that the VM is running with:
+```
+    PROJECT=elife-xpub vagrant status
+```
+
+The the deployment can be re-run with (if the VM is already running):
```
Comment (giorgiosironi) [inline local-development.md]: `Then`

File: local-development.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+Also deployments can be tested from within the VM using:
+```
+    sudo salt-call state.highstate
+```
+
+You can tear down all containers that have been run with docker-compose from the folder `/srv/elife-xpub` using:
+```
+    docker-compose down -v
+```
```
Comment (giorgiosironi) [inline local-development.md:66]: (this will delete the database contents)

Comment (giorgiosironi) [reply local-development.md:66]: (not just the containers but also volumes)

**Comment by diversemix** [comment]
@giorgiosironi Thanks for the feedback - ready to review again now.

File: local-development.md (hunk 5 above)
Comment (giorgiosironi) [reply local-development.md:66]: (resolved)

File: local-development.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
+# Local Deployment Tesing
+
+At times it may be necessary to change the way the deployment works, the most convenient way of doing this is to make changes locally. This avoids having to setup and wait for external services. This file contains some details on how to do this development locally.
```
Comment (giorgiosironi) [inline local-development.md:3]: A link to `elife-xpub-formula` would place this in context: this procedure helps with modifying that repository.

**Review by giorgiosironi — APPROVED** [review]

**Pull request merged**